fleetspeak/src/server/https/client_certificate.go

```diff
@@ -0,0 +1,52 @@
+package https
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// GetClientCert returns the client certificate from either the request header or TLS connection state.
+func GetClientCert(req *http.Request, hn string) (*x509.Certificate, error) {
+	if hn != "" {
+		return getCertFromHeader(hn, req.Header)
+	} else {
+		return getCertFromTLS(req)
+	}
+}
+
+func getCertFromHeader(hn string, rh http.Header) (*x509.Certificate, error) {
+	headerCert := rh.Get(hn)
+	if headerCert == "" {
+		return nil, errors.New("no certificate found in header")
+	}
```

Comment:
can we add logic here for also supporting envoy LB?
```
By=http://frontend.lyft.com;Hash=468ed33be74eee6556d90c0149c1309e9ba61d6425303443c0748a02dd8de688;Subject="/C=US/ST=CA/L=San Francisco/OU=Lyft/CN=Test Client";URI=http://testclient.lyft.com;Cert=<url encoded pem>"
```
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-client-cert

```suggestion
	headerCert := pemFromheader(rh.Get(hn))
	if headerCert == "" {
		return nil, errors.New("no certificate found in header")
	}
```
```golang
func pemFromHeader(header string) {
	if strings.Contains(header, "Cert=") {
                // split ";" and with "=" and get return the value for "Cert"    
        }
        return header;
}
```

Reply:
I think something like this should work: https://github.com/wozz/fleetspeak/commit/96eae48631cafe830dd69d15099a0ca27148a2e8

Reply:
Hi @dferreiralopessc, I definitely agree that the Envoy changes should go in. @wozz Can you make a separate PR for this?

Reply:
Pinging @torsm for situational awareness.